### ISSUE TITLE
reset connection on any exception

### ISSUE THREAD

**Issue by se-m** [uken/fluent-plugin-elasticsearch#213]
I offer a patch that helps with the problem #208
I applied the patch on my server - it works. Problem has gone.
The idea is to reset connection on any exception.
```ruby
  def send_bulk(data)
    retries = 0
    begin
      client.bulk body: data
    rescue *client.transport.host_unreachable_exceptions => e
      if retries < 2
        retries += 1
        @_es = nil
        log.warn "Could not push logs to Elasticsearch, resetting connection and trying again. #{e.message}"
        sleep 2**retries
        retry
      end
      raise ConnectionFailure, "Could not push logs to Elasticsearch after #{retries} retries. #{e.message}"
    rescue Exception => e
      @_es = nil
      raise e
    end #/begin
  end
```

**Comment by pitr** [comment]
I think this is a good idea. A few comments:

1. `Fluent::ObjectBufferedOutput` (from which this plugin inherits) already provides retry mechanism, perhaps there's a way to override its behaviour after it gives up.
2. This should be a config that's disabled by default, so as not to impact existing setups.

**Comment by se-m** [comment]
1. When it gives up, it simply cleans the queue. (retry mechanism in Fluent::BufferedOutput)
2. Ok

**Comment by cosmo0920** [comment]
Already included in v1.9 and v2.0 series. Closing.
